pkg/controller/pv.go

```diff
@@ -100,60 +103,27 @@ func (r *ReconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 	logger = logger.WithValues(consts.VolumeNameLabel, azVolumeName)
 	ctx = logr.NewContext(ctx, logger)
 
-	// PV is deleted
-	if deleteRequested, deleteAfter := objectDeletionRequested(&pv); deleteRequested {
-		if deleteAfter > 0 {
-			return reconcileAfter(deleteAfter, request.Name, r.controllerRetryInfo)
-		}
-		if err := r.cachedClient.Get(ctx, types.NamespacedName{Namespace: r.objectNamespace, Name: azVolumeName}, &azVolume); err != nil {
-			// AzVolume doesn't exist, so there is nothing for us to do
-			if errors.IsNotFound(err) {
-				return reconcileReturnOnSuccess(pv.Name, r.controllerRetryInfo)
-			}
-			// getting AzVolume failed, for unknown reason, we requeue and retry deleting it on the next cycle
-			logger.Error(err, "failed to get AzVolume")
-			return reconcileReturnOnError(ctx, &pv, "get", err, r.controllerRetryInfo)
-
-		}
-		// AzVolume does exist and needs to be deleted
-		// add annotation to mark AzVolumeAttachment cleanup
-		updateFunc := func(obj client.Object) error {
-			azv := obj.(*azdiskv1beta2.AzVolume)
-			azv.Status.Annotations = azureutils.AddToMap(azv.Status.Annotations, consts.PreProvisionedVolumeCleanupAnnotation, "true")
-			return nil
-		}
-		if _, err := azureutils.UpdateCRIWithRetry(ctx, nil, r.cachedClient, r.azClient, &azVolume, updateFunc, consts.NormalUpdateMaxNetRetry, azureutils.UpdateCRIStatus); err != nil {
-			return reconcileReturnOnError(ctx, &pv, "delete", err, r.controllerRetryInfo)
-		}
-
-		if err := r.azClient.DiskV1beta2().AzVolumes(r.objectNamespace).Delete(ctx, azVolumeName, metav1.DeleteOptions{}); err != nil {
-			logger.Error(err, "failed to set the deletion timestamp for AzVolume")
-			return reconcileReturnOnError(ctx, &pv, "delete", err, r.controllerRetryInfo)
-		}
-		// deletion timestamp is set and AzVolume reconcliler will handle the delete request.
-		// For pre-provisioned case when only PV is deleted, we will only be deleting the CRI as VolumeDeleteRequestAnnotation will not be set.
-		// The volume itself will not be deleted.
-		logger.Info("Deletion timestamp for AzVolume set")
-		return reconcileReturnOnSuccess(pv.Name, r.controllerRetryInfo)
-	}
-
 	// PV exists but AzVolume doesn't
+	deleteRequested, _ := objectDeletionRequested(&pv)
```

Comment:
It doesn't appear we are using `deleteRequested` other than in the following `if` statement scope. Can we move the check to the point of use (i.e. line 115 below)?